File: 08.BrainTwist脑筋急转弯/leetcode_3471_largestInteger.py
from typing import List
from collections import Counter

class Solution:
    def largestInteger(self, nums: List[int], k: int) -> int:
        if k == 1:
            return max([k for k,v in Counter(nums).items() if v==1]+[-1])
        if len(nums) == k:
            return max(nums)

        res = -1
        if nums.count(nums[0]) == 1:
            res = max(res, nums[0])
        if nums.count(nums[-1]) == 1:
            res = max(res, nums[-1])
        return res

if __name__ == "__main__":
    s = Solution()
    nums = [3,9,2,1,7]
    k = 3
    print(s.largestInteger(nums, k))

File: 08.BrainTwist脑筋急转弯/leetcode_3471_largestInteger.cpp
#include <iostream>
#include <vector>
#include <unordered_map>
#include <algorithm>
using namespace std;

class Solution {
public:
    int largestInteger(vector<int>& nums, int k) {
        if (k == 1){
            unordered_map<int, int> cnt;
            for (int i : nums) cnt[i]++;
            int res = -1;
            for (auto& kv : cnt) res = kv.second==1 ? max(res, kv.first) : res;
            return res;
        }
        if (nums.size() == k){
            return *max_element(nums.begin(), nums.end());
        }

        int res = -1;
        if (count(nums.begin(), nums.end(), nums[0]) == 1){
            res = max(res, nums[0]);
        }
        if (count(nums.begin(), nums.end(), nums.back()) == 1){
            res = max(res, nums.back());
        }
        return res;
    }
};

int main(){
    Solution s;
    vector<int> nums = {3,9,2,1,7};
    int k = 3;
    cout << s.largestInteger(nums, k) << endl;
}
